packages/server/modules/core/tests/embedTokens.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { TokenResourceIdentifierType } from '@/modules/core/domain/tokens/types'
+import { AllScopes } from '@/modules/core/helpers/mainConstants'
+import {
+  createRandomEmail,
+  createRandomPassword
+} from '@/modules/core/helpers/testHelpers'
+import {
+  BasicTestWorkspace,
+  createTestWorkspace
+} from '@/modules/workspaces/tests/helpers/creation'
+import { BasicTestUser, createTestUser } from '@/test/authHelper'
+import {
+  CreateEmbedTokenDocument,
+  GetActiveUserDocument,
+  GetProjectDocument,
+  GetWorkspaceDocument
+} from '@/test/graphql/generated/graphql'
+import {
+  createTestContext,
+  testApolloServer,
+  TestApolloServer
+} from '@/test/graphqlHelper'
+import { BasicTestStream, createTestStream } from '@/test/speckle-helpers/streamHelper'
+import { Roles, Scopes } from '@speckle/shared'
+import { expect } from 'chai'
+
+describe('Embed tokens', () => {
+  const adminUser: BasicTestUser = {
+    id: '',
+    name: 'John Speckle',
+    email: createRandomEmail(),
+    password: createRandomPassword()
+  }
+
+  const workspace: BasicTestWorkspace = {
+    id: '',
+    ownerId: '',
+    name: 'My Workspace',
+    slug: ''
+  }
+
+  const projectA: BasicTestStream = {
+    id: '',
+    ownerId: '',
+    name: 'My Project'
+  }
+  const projectB: BasicTestStream = {
+    id: '',
+    ownerId: '',
+    name: 'My Project 2'
+  }
+
+  let apollo: TestApolloServer
+
+  before(async () => {
+    await createTestUser(adminUser)
+
+    await createTestWorkspace(workspace, adminUser)
+
+    projectA.workspaceId = workspace.id
+    projectB.workspaceId = workspace.id
+
+    await createTestStream(projectA, adminUser)
+    await createTestStream(projectB, adminUser)
+
+    const adminApollo = await testApolloServer({
+      context: await createTestContext({
```

Comment:
its not necessary to manually build the entire context, you can just pass in `authUserId` which will authenticate the context w/ that user id

Reply:
This is meant to use the token explicitly and this was how I managed to get it to consider scope/resource access limits. Is there a better way to set this up if I want to emulate a client with a token I have?

Reply:
authUserId does set default scopes/roles, that's true, so if you want to customize those then yea you have to create a manual test context